app/models/teacher_set.rb

```diff
@@ -102,6 +102,41 @@ def holds_for_user(user, hold_id)
     end
   end
 
+  
+  def teacher_set_availability(available_copies)
+    available_copies.to_i > 0 ? AVAILABLE : UNAVAILABLE
+  end
+
+
+  def calculate_available_copies(status, quantity, current_user, hold_id)
+    if status == 'cancelled'
```

Comment:
Would be worth adding a comment here saying that `quantity` is not set for cancellations, so must be calculated